senseobjects/app.go

```diff
@@ -123,6 +131,145 @@ func (app *App) setBookmarkList(sessionState SessionState, bl *BookmarkList) {
 	app.bookmarkList = bl
 }
 
+// GetVariableList create or return existing variable list session object
+func (app *App) GetVariableList(sessionState SessionState, actionState *action.State) (*VariableList, error) {
```

Comment:
This function could benefit from being made generic with the help of an interface and/or a function argument.

Reply:
was thinking about it, but for most cases we need / or will need when accessing details, specific structs for the layout and properties(like e.g. sheetlist and bookmarklist), so would be little gain in the end

Reply:
or you mean GetSessionObject(sdslfsdf) and we autodetect the rest? semiprepared for that by separating the current session objects to its own struct. Would agree a "sessionobjecthandler" or something shoule be implemented at some point.

Reply:
would only work for these three and only while they use the GenericLayout and GenericProperties, as soon as we have to make any of them their own type the need to be broken out again, since this has been the case with the other session objects I consider it a likely scenerio to happen. Also remember

layout := MyLayout{}
myfunc(&layout)

func myfunc(layout interface{}{ 
...
json.Unmarshal(data, layout)
...
}
will make layout into a map[string] instead of a MyLayout

Reply:
that said, some code could be shared, but would rather make something that covers "all" sessionobjects in that case, which would be a bigger rework then I wanted to do now.